File: VB/S170863/DrawEditorHelper.vb
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraVerticalGrid.Events

Namespace WindowsFormsApplication1
    Public NotInheritable Class DrawEditorHelper

        Private Sub New()
        End Sub

        Public Shared Sub DrawEdit(ByVal g As Graphics, ByVal edit As RepositoryItem, ByVal r As Rectangle, ByVal value As Object)
            Dim info As BaseEditViewInfo = edit.CreateViewInfo()
            info.EditValue = value
            info.Bounds = r
            info.CalcViewInfo(g)
            Dim args As New ControlGraphicsInfoArgs(info, New GraphicsCache(g), r)
            edit.CreatePainter().Draw(args)
            args.Cache.Dispose()
        End Sub

        Public Shared Sub DrawRowInplaceEditor(ByVal e As CustomDrawRowHeaderCellEventArgs, ByVal item As RepositoryItem, ByVal value As Object)
            Dim targetRect As Rectangle = e.Bounds
            DrawEdit(e.Graphics, item, targetRect, value)
        End Sub
    End Class
End Namespace

File: CS/S170863/DrawEditorHelper.cs
using DevExpress.Utils.Drawing;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraVerticalGrid.Events;
using System.Drawing;

namespace WindowsFormsApplication1
{
    public static class DrawEditorHelper {
        public static void DrawEdit(Graphics g, RepositoryItem edit, Rectangle r, object value) {
            BaseEditViewInfo info = edit.CreateViewInfo();
            info.EditValue = value;
            info.Bounds = r;
            info.CalcViewInfo(g);
            ControlGraphicsInfoArgs args = new ControlGraphicsInfoArgs(info, new GraphicsCache(g), r);
            edit.CreatePainter().Draw(args);
            args.Cache.Dispose();
        }

        public static void DrawRowInplaceEditor(CustomDrawRowHeaderCellEventArgs e, RepositoryItem item, object value) {
            Rectangle targetRect = e.Bounds;
            DrawEdit(e.Graphics, item, targetRect, value);
        }
    }
}
